lib/matplotlib/offsetbox.py

```diff
@@ -1505,18 +1506,48 @@ def __init__(self, ref_artist, use_blit=False):
         if not ref_artist.pickable():
             ref_artist.set_picker(True)
         self.got_artist = False
+        self._hover = False
         self._use_blit = use_blit and self.canvas.supports_blit
         self.cids = [
             self.canvas.callbacks._connect_picklable(
                 'pick_event', self.on_pick),
             self.canvas.callbacks._connect_picklable(
                 'button_release_event', self.on_release),
+            self.canvas.callbacks._connect_picklable(
+                'motion_notify_event', self.on_motion),
         ]
 
     # A property, not an attribute, to maintain picklability.
     canvas = property(lambda self: self.ref_artist.figure.canvas)
 
     def on_motion(self, evt):
+        # Only check if the widget lock is available, setting it would prevent
+        # picking.
+        if not (
+            self._check_still_parented() and self.canvas.widgetlock.available(self)
+        ):
+            return
+
+        picker = self.ref_artist.get_picker()
+        if callable(picker):
+            inside, _ = picker(self, evt)
+        else:
+            inside, _ = self.ref_artist.contains(evt)
+
+        # If the mouse is moving quickly while dragging, it may leave the artist,
+        # but should still use the move cursor.
+        if inside or self.got_artist:
+            # Ideally, this should use an open hand cursor on hover, and a closed
+            # hand when dragging, but those cursors are not natively provided by
+            # all platforms.
```

Comment:
Is there a way to try/except that behavior if it is available?

On macosx, when I go to pan the image if I'm hovering over the image it is an open hand, but when I click down it is a closed hand. With this there is an inconsistency for the legend compared to that. Actually when I first click down over the legend the hand does close, but then it immediately opens up again.

Reply:
Is that on the macosx backend, or a different backend on macosx? I don't have a mac to test on, but if it's a different backend, it might be due to `set_cursor()` resetting the cursor, so I'd need to see if changing the the condition to `if (inside or self.got_artist) and not self._hover` works instead. Otherwise, the behaviour should be identical to the pan tool, unless there's some special code for just that on mac.

Reply:
It looks like it is only with the macosx backend, and it is handled internally to go from closed/open-hand.
https://github.com/matplotlib/matplotlib/blob/205e9075eb50bd7f9efade89391863354032aeea/src/_macosx.m#L1524-L1525

Note:
It looks like QT does have the option for closed/open hand: https://doc.qt.io/qt-5/qcursor.html
But I couldn't find a similar option for tk.

Your suggestion for the if condition seems like it might work with avoiding sending too many `set_cursor()`s, I think that suggestion is similar to what is going on in pan keeping track of last_cursor.
https://github.com/matplotlib/matplotlib/blob/205e9075eb50bd7f9efade89391863354032aeea/lib/matplotlib/backend_bases.py#L3128-L3131